deepchecks/nlp/utils/text_properties.py

```diff
@@ -25,22 +26,82 @@
 from nltk import sent_tokenize, word_tokenize
 from typing_extensions import TypedDict
 
+from deepchecks.core.errors import DeepchecksValueError
 from deepchecks.nlp.utils.text import remove_punctuation
 from deepchecks.utils.function import run_available_kwargs
 from deepchecks.utils.ipython import create_progress_bar
 
 __all__ = ['calculate_builtin_properties']
 
-
 MODELS_STORAGE = pathlib.Path(__file__).absolute().parent / '.nlp-models'
 FASTTEXT_LANG_MODEL = 'https://dl.fbaipublicfiles.com/fasttext/supervised-models/lid.176.bin'
+DEFAULT_SENTENCE_SAMPLE_SIZE = 300
+properties_cache = {}
+
+
+def _sample_for_property(text: str, mode: str = 'words', limit: int = 10000, return_as_list=False,
+                         random_seed: int = 42) -> Union[str, List[str]]:
+    """Get a sample a single text sample for a text property.
+
+    Parameters
+    ----------
+    text : str
+        The text to sample from.
+    mode : str, default 'words'
+        The mode to sample in. Can be either 'words' or 'sentences'.
+    limit : int, default 10000
+        The maximum number of words or sentences to sample.
+    """
+    np.random.seed(random_seed)
+    if pd.isna(text):
```

Comment:
not sure why I think that but isnull may be more complete

Reply:
Honestly just following the previous removed code:
![image](https://github.com/deepchecks/deepchecks/assets/92314933/1f22c860-7c2e-406e-8149-404551700c62)